airflow/contrib/hooks/bigquery_hook.py

```diff
@@ -1718,6 +1717,75 @@ def get_datasets_list(self, project_id=None):
 
         return datasets_list
 
+    def get_dataset_tables_list(self, dataset_id, project_id=None, table_prefix=None, max_results=None):
```

Comment:
What do you think about adding the @GoogleCloudBaseHook.fallback_to_default_project_id``decorator? This will allow you to delete the repeating code, but also add validation - this field is required.

Reference: https://github.com/apache/airflow/pull/5546/files#r307964979